### ISSUE TITLE
C# generator compiler errors on enums when enum is a number

### ISSUE THREAD

**Issue by jsmarsch** [NSwag/NSwag#172]
I used NSwag Studio to generate c# code for a swagger where there was an enumeration defined, and the values of the enumeration were numbers.  

That causes a problem,  because a number is not a valid identifier in c#.

Here's an example of the generator output:

```
    public enum ClassCode1
    {
        0562 = 0,
        0532 = 1,
        7391 = 2,
        01139 = 3,
    }
```

The REST service expects values like "0562", but we can't use that as the name of a value for an enum.

I compared this to the output of https://github.com/swagger-api/swagger-codegen and I noticed that generator prefixes numeric identifiers with an underscore, so it would produce:
```
    public enum ClassCode1
    {
        _0562 = 0,
        _0532 = 1,
        _7391 = 2,
        _01139 = 3,
    }
```

Perhaps NSwag should do something similar?

**Comment by rsuter** [comment]
Yes, good find :)

**Comment by jsmarsch** [comment]
Thanks -- I realized just now that I left something out.  The other generator also generates EnumMember attributes on those enums so that they serialize correctly.  So, the class code looks like this:
(this time a full copy/paste, not free-handing it):

```
        [JsonConverter(typeof(StringEnumConverter))]
        public enum ClassCodeEnum {
            
            [EnumMember(Value = "0562")]
            _0562,
            
            [EnumMember(Value = "0532")]
            _0532
        }

```

**Comment by rsuter** [comment]
I currently don't use the C# generator. This is why I didn't find all these edge cases yet... Will fix this in the next days...

**Comment by rsuter** [comment]
Btw: Why are you using NSwag over swagger-codegen?

**Comment by rsuter** [comment]
Do you use NSwag generated swagger or external?

**Comment by jsmarsch** [comment]
Hi rsuter:  Answers to your questions:

I'm just getting started with Swagger, and we are a .net shop, We are consuming some REST services that have Swagger set up for them (not building any right now).  I'm not sure what they used to generate the swagger.

So, my first attempt at generating code was to use the "Generate Client Code" option from the swagger  online editor here: http://editor.swagger.io/#/.  When it didn't compile, I started looking around for alternatives and found NSwag.  I liked that there is was a C# API in addition to the command line and GUI options, so I tried it out.

I generated the NSwag-generated code with the NSwag GUI

Some more searching led me to swagger-codegen, so I pulled down the .jar file, and tried it.  Oddly, while the Web Site produced code that did not compile, the .jar produced code that will.  (the code is a little odd though -- c# is probably not the primary language used by the person who wrote the generator)

In all honesty, my own needs are pretty simple -- I want to generate really "dumb" DTO's.  For my purposes, I would actually rather just have string properties than "real" enums for the DTO's that I am going to use (perhaps with some constants for the known values.

I may just end up hand-coding the DTO's, but I had hoped to generate them because they are rather large.
